src/interfaces/table.ts

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 OpenStax Poland
+// Licensed under the MIT license. See LICENSE file in the project root for
+// full license text.
+
+import { Element } from 'slate'
+
+/** Collection of media elements */
+export interface Table extends Element {
+    type: 'table'
+}
+
+export const Table = {
+    /** Check if value of unknown type is a collection of media elements */
+    isTable(value: unknown): value is Table {
+        return Element.isElement(value) && value.type === 'table'
+    },
+}
+
+export interface TableTitle extends Element {
+    type: 'table_title'
+}
+
+export const TableTitle = {
+    /** Check if value of unknown type is a collection of media elements */
+    isTableTitle(value: unknown): value is TableTitle {
+        return Element.isElement(value) && value.type === 'table_title'
+    },
+}
+
+export interface TableTgroup extends Element {
+    type: 'table_tgroup'
+}
+
+export const TableTgroup = {
+    /** Check if value of unknown type is a collection of media elements */
+    isTableTgroup(value: unknown): value is TableTgroup {
+        return Element.isElement(value) && value.type === 'table_tgroup'
+    },
+}
+
+export interface TableTbody extends Element {
+    type: 'table_tbody'
+}
+
+export const TableTbody = {
+    /** Check if value of unknown type is a collection of media elements */
+    isTableTbody(value: unknown): value is TableTbody {
+        return Element.isElement(value) && value.type === 'table_tbody'
+    },
+}
+
+export interface TableRow extends Element {
+    type: 'table_row'
+}
+
+export const TableRow = {
+    /** Check if value of unknown type is a collection of media elements */
+    isTableRow(value: unknown): value is TableRow {
+        return Element.isElement(value) && value.type === 'table_row'
+    },
+}
+
+export interface TableEntry extends Element {
+    type: 'table_entry'
+}
+
+export const TableEntry = {
+    /** Check if value of unknown type is a collection of media elements */
+    isTableEntry(value: unknown): value is TableEntry {
+        return Element.isElement(value) && value.type === 'table_entry'
+    },
+}
```

Comment:
Why is code for tables included in this PR?

Reply:
I needed an interface for a table. Make it a different PR?

Reply:
We don't know yet how to handle tables, so it shouldn't be added at all (tables are currently implemented in a ‘temporary’ way in adaptarr-front).